cmd/otel-allocator/allocation/per_node_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package allocation
+
+import (
+	"testing"
+
+	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/target"
+
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/assert"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var loggerPerNode = logf.Log.WithName("unit-tests")
+
+// Tests that two targets with the same target url and job name but different label set are both added.
+func TestAllocationPerNode(t *testing.T) {
```

Comment:
I'd like to see a few more tests for edge cases. No collectors, no targets, those kinds of conditions. You can probably copy those from consistent-hashing. Ideally we'd have a strategy-agnostic set of tests, but I'm not going to ask that you add those to this (already large and complex) PR.

Reply:
Added tests as suggested.